rskj-core/src/main/java/co/rsk/core/types/bytes/BytesSlice.java

```diff
@@ -37,17 +88,30 @@ public interface BytesSlice extends HexPrintableBytes {
      * greater than or equal to <tt>original.length - from</tt>.  The length
      * of the returned array will be <tt>to - from</tt>.
      *
+     * <p>
+     * Note: this method mimics behaviour of {@link Arrays#copyOfRange(Object[], int, int)}
+     *
      * @param from the initial index of the range to be copied, inclusive
      * @param to the final index of the range to be copied, exclusive.
      *     (This index may lie outside the array.)
      * @return a new array containing the specified range from the original array,
      *     truncated or padded with zeros to obtain the required length
-     * @throws ArrayIndexOutOfBoundsException if {@code from < 0}
+     * @throws IndexOutOfBoundsException if {@code from < 0}
      *     or {@code from > original.length}
      * @throws IllegalArgumentException if <tt>from &gt; to</tt>
-     * @throws NullPointerException if <tt>original</tt> is null
      */
-    byte[] copyArrayOfRange(int from, int to);
+    default byte[] copyArrayOfRange(int from, int to) {
+        if (from < 0 || from > length()) {
+            throw new IndexOutOfBoundsException("invalid 'from': " + from);
+        }
+        int newLength = to - from;
+        if (newLength < 0) {
+            throw new IllegalArgumentException(from + " > " + to);
+        }
+        byte[] copy = new byte[newLength];
```

Comment:
should `copy` array length also be `Math.min(length() - from, newLength)`? in other case it may contain zero bytes at the end which are not present in original array

Reply:
@VaWheel , the idea here was to mimic the behaviour of `Arrays#copyOfRange(Object[], int, int)` (pls see [this](https://github.com/rsksmart/rskj/pull/2647/files#diff-9d2592af93c59be41eba3306a1edfd3a80162db11a361f4a15bf59baa87ffe04R92) and [this](https://github.com/rsksmart/rskj/pull/2647/files#diff-9d2592af93c59be41eba3306a1edfd3a80162db11a361f4a15bf59baa87ffe04R97-R98) javadoc's)